pkg/libartifact/store/store.go

```diff
@@ -0,0 +1,375 @@
+	if len(nameOrDigest) == 0 {
+		return nil, ErrEmptyArtifactName
+	}
+	artifacts, err := as.getArtifacts(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	inspectData, _, err := artifacts.GetByNameOrDigest(nameOrDigest)
+	return inspectData, err
+}
+
+// List artifacts in the local store
+func (as ArtifactStore) List(ctx context.Context) (libartifact.ArtifactList, error) {
+	return as.getArtifacts(ctx, nil)
+}
+
+// Pull an artifact from an image registry to a local store
+func (as ArtifactStore) Pull(ctx context.Context, name string, opts libimage.CopyOptions) error {
+	if len(name) == 0 {
+		return ErrEmptyArtifactName
+	}
+	srcRef, err := alltransports.ParseImageName(fmt.Sprintf("docker://%s", name))
+	if err != nil {
+		return err
+	}
+	destRef, err := layout.NewReference(as.storePath, name)
+	if err != nil {
+		return err
+	}
+	copyer, err := libimage.NewCopier(&opts, as.SystemContext, nil)
+	if err != nil {
+		return err
+	}
+	_, err = copyer.Copy(ctx, srcRef, destRef)
+	if err != nil {
+		return err
+	}
+	return copyer.Close()
+}
+
+// Push an artifact to an image registry
+func (as ArtifactStore) Push(ctx context.Context, src, dest string, opts libimage.CopyOptions) error {
+	if len(dest) == 0 {
+		return ErrEmptyArtifactName
+	}
+	destRef, err := alltransports.ParseImageName(fmt.Sprintf("docker://%s", dest))
+	if err != nil {
+		return err
+	}
+	srcRef, err := layout.NewReference(as.storePath, src)
+	if err != nil {
+		return err
+	}
+	copyer, err := libimage.NewCopier(&opts, as.SystemContext, nil)
+	if err != nil {
+		return err
+	}
+	_, err = copyer.Copy(ctx, srcRef, destRef)
+	if err != nil {
+		return err
+	}
+	return copyer.Close()
+}
+
+// Add takes one or more local files and adds them to the local artifact store.  The empty
+// string input is for possible custom artifact types.
+func (as ArtifactStore) Add(ctx context.Context, dest string, paths []string, _ string) (*digest.Digest, error) {
+	if len(dest) == 0 {
+		return nil, ErrEmptyArtifactName
+	}
+
+	artifactManifestLayers := make([]specV1.Descriptor, 0)
+
+	// Check if artifact already exists
+	artifacts, err := as.getArtifacts(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if artifact exists; in GetByName not getting an
+	// error means it exists
+	if _, _, err := artifacts.GetByNameOrDigest(dest); err == nil {
+		return nil, fmt.Errorf("artifact %s already exists", dest)
+	}
+
+	ir, err := layout.NewReference(as.storePath, dest)
+	if err != nil {
+		return nil, err
+	}
+
+	imageDest, err := ir.NewImageDestination(ctx, as.SystemContext)
+	if err != nil {
+		return nil, err
+	}
+	defer imageDest.Close()
+
+	for _, path := range paths {
+		// get the new artifact into the local store
+		newBlobDigest, newBlobSize, err := layout.PutBlobFromLocalFile(ctx, imageDest, path)
+		if err != nil {
+			return nil, err
+		}
+		mediaType := runtime.DefaultMime
```

Comment:
Is this actually the correct import?
This is the only reference for github.com/go-openapi/runtime in our direct code.
In particular you only use it to define `application/octet-stream`, I am not sure if we should use it for that. The name DefaultMime implies that might be suspect to change and then we would not easily notice.
would it not be better to dinfe this as our own constant here?

Reply:
yes, nalin said so.

Reply:
I rescind that recommendation, though tbh I don't remember suggesting it.